### PR TITLE
INT-2422 GraphQL Pagination

### PR DESCRIPTION
**Pull request by VDubber**
This continues the refactor of GraphQL Pagination.
- Moves remaining GraphQL queries to new pattern
- Removes existing  now unused code
- Introduces types to help new queries to be built

### CODE REVIEW

File: src/client/GraphQLClient/collaboratorQueries/RepoCollaboratorsQuery.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  responseData,
+  iteratee,
+) => {
+  const rateLimit = responseData.rateLimit;
+  const collaboratorEdges = responseData.repository?.collaborators?.edges ?? [];
+
+  for (const edge of collaboratorEdges) {
+    const node = edge.node;
+
+    const collaborator: Collaborator = {
```
Comment (VDubber) [inline src/client/GraphQLClient/collaboratorQueries/RepoCollaboratorsQuery.ts:72]: Verify that node is defined and not and empty object

File: src/client/GraphQLClient/collaboratorQueries/RepoCollaboratorsQuery.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      id: node.id,
+      name: node.name,
+      login: node.login,
+      permission: edge.permission,
+      repositoryId: responseData.repository?.id,
```
Comment (VDubber) [inline src/client/GraphQLClient/collaboratorQueries/RepoCollaboratorsQuery.ts:77]: remove ?

File: src/client/GraphQLClient/issueQueries/IssuesQuery.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  if (!responseData) {
+    throw new Error('responseData param is required');
+  }
+
+  const rateLimit = responseData.rateLimit;
+  const issueEdges = responseData.search.edges;
+
+  for (const edge of issueEdges) {
+    const issue = edge.node;
+    if (Object.keys(issue).length === 0) {
```
Comment (VDubber) [inline src/client/GraphQLClient/issueQueries/IssuesQuery.ts:127]: move to a utility?

File: src/client/GraphQLClient/issueQueries/IssuesQuery.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+) => {
+  return paginate(
+    queryParams,
+    iteratee,
+    execute,
+    buildQuery,
+    processResponseData,
+    (queryState, issuesFetched) =>
+      (!queryState?.issues?.hasNextPage ?? true) ||
+      issuesFetched >= MAX_FETCHES_PER_EXECUTION,
```
Comment (VDubber) [inline src/client/GraphQLClient/issueQueries/IssuesQuery.ts:167]: rename MAX_FETCHES_PER_EXECUTION to MAX_RESOURCES_PER_EXECUTION

File: src/client/GraphQLClient/repositoryQueries/OrgRepositoriesQuery.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  queryState?: QueryState,
+): ExecutableQuery => {
+  const query = `
+      query ($login: String!, $maxLimit: Int!, $repoCursor: String) {
+        organization(login: $login) {
+          id
+          repositories(first: $maxLimit, after: $repoCursor) {
+            nodes {
+              id
+					    ...repositoryFields
```
Comment (VDubber) [inline src/client/GraphQLClient/repositoryQueries/OrgRepositoriesQuery.ts:35]: formatting

File: src/client/OrganizationAccountClient.ts [hunk 6]
```diff
@@ -317,16 +217,16 @@ export default class OrganizationAccountClient {
     //issues and PRs are actually the same in the API
     //we just filter for is:issue instead of is:pr
     //and remove pr-specific children from the request
+    // TODO: SP -> investigate removing pr-specific children
```
Comment (VDubber) [inline src/client/OrganizationAccountClient.ts:220]: Consider/remove TODO

File: src/config.ts [hunk 7]
```diff
@@ -153,8 +153,6 @@ export function sanitizeConfig(config: IntegrationConfig) {
       'https://api.github.com',
   );
 
-  config.installationId = config.installationId ?? config.githubInstallationId;
```
Comment (VDubber) [inline src/config.ts:156]: Remove unnecessary null handling. Integration project now handles this.

File: src/client.ts [hunk 8]
```diff
@@ -95,11 +95,15 @@ export class APIClient {
...
-      await this.accountClient.getMembers();
-    for (const member of members) {
-      await iteratee(member);
-    }
+
+    const { rateLimitConsumed } = await this.accountClient.iterateOrgMembers(
+      iteratee,
+    );
+
+    this.logger.info(
```
Comment (VDubber) [inline src/client.ts:103]: These logs are not helpful :) Consolidate after each step?

File: src/client/GraphQLClient/issueQueries/IssuesQuery.ts (hunk 3 above)
Comment (VDubber) [reply src/client/GraphQLClient/issueQueries/IssuesQuery.ts:125]: Done

File: src/client/GraphQLClient/repositoryQueries/OrgRepositoriesQuery.ts (hunk 5 above)
Comment (VDubber) [reply src/client/GraphQLClient/repositoryQueries/OrgRepositoriesQuery.ts:35]: Fixed

File: src/client/GraphQLClient/issueQueries/IssuesQuery.ts (hunk 4 above)
Comment (VDubber) [reply src/client/GraphQLClient/issueQueries/IssuesQuery.ts:165]: Done

File: src/client.ts [hunk 9]
```diff
@@ -113,15 +124,14 @@ export class APIClient {
     if (!this.accountClient) {
       await this.setupAccountClient();
     }
-    const teams: OrgTeamQueryResponse[] = await this.accountClient.getTeams();
-    for (const team of teams) {
-      await iteratee(team);
-    }
+
+    await this.accountClient.iterateTeams(iteratee);
```
Comment (adam-in-ict) [inline src/client.ts:128]: Does this call not include the rate limit consumed in its response like other similar accountClient calls?

File: src/client.ts [hunk 10]
```diff
@@ -343,17 +365,15 @@ export class APIClient {
...
       await this.setupAccountClient();
     }
-    const collaborators = await this.accountClient.getRepoCollaborators(
-      repoName,
-    );
-    for (const collab of collaborators) {
-      await iteratee(collab);
-    }
+
+    await this.accountClient.iterateRepoCollaborators(repoName, iteratee);
```
Comment (adam-in-ict) [inline src/client.ts:369]: Same question regarding getting rateLimit data from this call similar to other accountClient calls.

File: src/client/GraphQLClient/README.md [hunk 11]
```diff
@@ -1,7 +1,8 @@
 ## Creating GraphQL Queries
 
 When creating a new query, consider the query as a new domain being added to the
-project.
+project. To prevent needing to setup a new GitHub organization that has test
+data, contact samuel.poulton@jupiterone.com to gain access to a test GitHub org.
```
Comment (adam-in-ict) [inline src/client/GraphQLClient/README.md:5]: Any concern with this being in a public repo?  I imagine we've only got our test GitHub account set up for internal testing/use, correct?

File: src/client.ts (hunk 9 above)
Comment (VDubber) [reply src/client.ts:128]: It does! I just skipped it. I'll get that patched up

File: src/client/GraphQLClient/README.md (hunk 11 above)
Comment (VDubber) [reply src/client/GraphQLClient/README.md:5]: Good thought. I'll remove my email and direct them to https://github.com/j1-ingest 
Along with add some details on that Org's profile page. From there, someone should be able to find me or maybe support

**Review by adam-in-ict — APPROVED** [review]

File: src/client/GraphQLClient/collaboratorQueries/RepoCollaboratorsQuery.ts (hunk 1 above)
Comment (VDubber) [reply src/client/GraphQLClient/collaboratorQueries/RepoCollaboratorsQuery.ts:76]: Done

**Pull request merged**